core/modules/image/image.field.inc

```diff
@@ -759,33 +822,153 @@ function image_field_formatter_view($entity_type, $entity, $field, $instance, $l
+        $item['attributes']['class'][] = 'align-' . $display['settings']['image_float'];
+      }
+
+      if (!empty($display['settings']['image_load'])) {
+        // Although available in Chromium, the 'auto' value is not mentioned in
+        // the specification. Since it may be subject to change, we recommend
+        // not to use it until it gets officially included.
+        if ($display['settings']['image_load'] != 'auto') {
+          $item['attributes']['loading'] = $display['settings']['image_load'];
+        }
+      }
+
+      if (isset($link_file)) {
+        $uri = array(
+          'path' => file_create_url($item['uri']),
+          'options' => array(),
+        );
+      }
+
+      $is_svg = image_is_svg($item['uri']);
+
+      // Add custom attributes only for SVG images.
+      if ($is_svg) {
+        $image_svg_dimensions = $display['settings']['image_svg_dimensions'];
 
-    if (!empty($display['settings']['image_load'])) {
-      // Although available in Chromium, the 'auto' value is not mentioned in
-      // the specification. Since it may be subject to change, we recommend not
-      // to use it until it gets officially included.
-      if ($display['settings']['image_load'] != 'auto') {
-        $item['attributes']['loading'] = $display['settings']['image_load'];
+        foreach ($image_svg_dimensions as $attribute => $attribute_value) {
+          if (!empty($attribute)) {
+            $item[$attribute] = $attribute_value;
+          }
+        }
       }
     }
 
-    if (isset($link_file)) {
-      $uri = array(
-        'path' => file_create_url($item['uri']),
-        'options' => array(),
+    if (!$is_svg || $display['settings']['image_svg_display']) {
+      $element[$delta] = array(
+        '#theme' => 'image_formatter',
+        '#item' => $item,
+        '#image_style' => $display['settings']['image_style'],
+        '#path' => isset($uri) ? $uri : '',
       );
     }
-    $element[$delta] = array(
-      '#theme' => 'image_formatter',
-      '#item' => $item,
-      '#image_style' => $display['settings']['image_style'],
-      '#path' => isset($uri) ? $uri : '',
-    );
+    else {
+      $svg_path = file_create_url($item['uri']);
+      $svg = file_get_contents($svg_path);
+      $svg = preg_replace(['/<\?xml.*\?>/i', '/<!DOCTYPE((.|\n|\r)*?)">/i'], '', $svg);
+      $svg_clean = filter_xss($svg, image_field_allowed_svg_tags());
+      $svg_clean = trim($svg_clean);
+
+      if (isset($uri)) {
+        $element[$delta] = array(
+          '#type' => 'link',
+          '#href' => $uri['path'],
+          '#title' => $svg_clean,
+          '#options' => array(
+            'html' => TRUE,
+          ),
+        );
+      }
+      else {
+        $element[$delta] = array(
+          '#type' => 'markup',
+          '#markup' => $svg_clean,
+        );
+      }
+    }
   }
 
   return $element;
 }
+
+/**
+ * Provides a list of allowed SVG tags.
+ */
```

Comment:
Could we add some documentation link to the comment - where this info's from and could get verified? In case, such a link is available, of course.

Reply:
https://www.w3.org/TR/SVG/eltindex.html

Reply:
Ah, many thanks...
I actually meant to add it to the doc block with an `@see` :grinning: